sample-apps/go-sample-app/collection/request_metrics.go

```diff
@@ -0,0 +1,110 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync/atomic"
+
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+)
+
+// requestBasedMetricCollector contains all the request based metric instruments.
+type requestBasedMetricCollector struct {
+	totalBytesSent syncint64.Counter
+	totalRequests  asyncint64.Counter
+	latencyTime    syncint64.Histogram
+	context        context.Context
```

Comment:
It is generally bad form to store a context in a struct.  Why does this need to be stored here?

Reply:
I removed context from the struct. It was an old attempt at context propagation but after understanding it further, i realized this was incorrect